dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/datasetreport/impl/DefaultDataSetReportService.java

```diff
@@ -316,13 +318,56 @@ else if ( formType.isSection() )
             tmpDataSet.getSections().add( section );
         }
 
-        return getSectionDataSetReport( tmpDataSet, period, unit, filters, selectedUnitOnly );
+        return getSectionDataSetReport( tmpDataSet, periods, unit, filters, selectedUnitOnly );
     }
 
     // -------------------------------------------------------------------------
     // Supportive methods
     // -------------------------------------------------------------------------
 
+    private String formatPeriods( List<Period> periods, I18nFormat format )
+    {
+        Collections.sort( periods );
+
+        List<String> formattedPeriods = periods.stream()
+            .map( p -> format.formatPeriod( p ) )
+            .collect( Collectors.toList() );
```

Comment:
I think there is a joiner for streams so you don't need to collect to list first and then join to string.
Also if you use streams anyhow you could just sort the stream instead - should be both more readable and with a bit of luck less work to do.

Reply:
Thanks, much more compact now. Done.